packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -0,0 +1,389 @@
+              packageQuickPickItems.push({
+                label: `${dep.name}`,
+                detail: "Required for enabling the emitter library.",
+                packageFullName: packageFullName,
+                picked: true,
+              });
+            }
+          }
+        } catch (err) {
+          logger.error(`Exception occurred when check dependency packages for ${p}.`);
+        }
+      }
+      return packageQuickPickItems;
+    },
+  );
+
+  if (installPackageQuickPickItems.length > 0) {
+    const selectedPackages = await vscode.window.showQuickPick(installPackageQuickPickItems, {
+      title: "Install or update libraries",
+      canPickMany: true,
+      placeHolder: "Here are libraries to install or update.",
+      ignoreFocusOut: true,
+    });
+    if (!selectedPackages || selectedPackages.length === 0) {
+      logger.info("No package selected. Generating Cancelled.", [], {
+        showOutput: true,
+        showPopup: true,
+      });
+      return;
+    }
+    /* npm install packages. */
+    if (selectedPackages.length > 0) {
+      const installPackages = selectedPackages.map((p) => p.packageFullName);
+      logger.info(`Install ${installPackages.join("\n\n")} under directory ${baseDir}`, [], {
+        showOutput: true,
+        showPopup: true,
+      });
+      const installResult = await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          title: "Installing packages...",
+          cancellable: false,
+        },
+        async () => {
+          try {
+            const npmInstallResult = await npmUtil.npmInstallPackages(installPackages, undefined);
+            if (npmInstallResult.exitCode !== 0) {
+              logger.error(
+                `Error occurred when installing packages.`,
+                [`${npmInstallResult.stderr}`],
+                {
+                  showOutput: true,
+                  showPopup: true,
+                },
+              );
+              return false;
+            } else {
+              return true;
+            }
+          } catch (err) {
+            logger.error(`Exception occurred when installing packages.`, [err], {
+              showOutput: true,
+              showPopup: true,
+            });
+            return false;
+          }
+        },
+      );
+      if (!installResult) {
+        logger.error(`Error occurred when installing packages. Generating Cancelled.`, [], {
+          showOutput: false,
+          showPopup: true,
+        });
+        return;
+      }
+    }
+  }
+
+  /* emit */
+  const cli = await resolveTypeSpecCli(baseDir);
+  if (!cli) {
+    logger.error(
+      "Cannot find TypeSpec CLI. Please install @typespec/compiler. Generating Cancelled.",
+      [],
+      {
+        showOutput: true,
+        showPopup: true,
+      },
+    );
+    return;
+  }
+  const outputDir = path.join(
+    baseDir,
+    selectedEmitter.emitterKind,
```

Comment:
please put the output folder under tsp-output as default. the "server", "client" is too general which user may already use in his project...

Reply:
@RodgeFu  We need consistent the output folder structure as the toDoApp. Arch will discuss the folder structure, and update accordingly when toDoApp folder structure finalized.

Reply:
@chunyu3, as discussed, you should not control the output folder yourself anymore. It should be what's configured in the tspconfig.yaml

Reply:
@allenjzhang We will honor the output folder in tspconfig.yaml if there is any, if there is no, we will give default output folder. This code show is out-of-date.